cf-custom-resources/lib/env-controller.js

```diff
@@ -100,25 +98,32 @@ const controlEnv = async function (
     }
     var updatedEnvStack = describeStackResp.Stacks[0];
     var params = JSON.parse(JSON.stringify(updatedEnvStack.Parameters));
-    var updated = false;
-    for (const param of params) {
-      for (const envParam of envParameters) {
-        if (param.ParameterKey === envParam) {
-          const [updatedParamValue, paramUpdated] = updateParameter(
-            requestType,
-            workload,
-            param.ParameterValue
-          );
-          param.ParameterValue = updatedParamValue;
-          updated = updated || paramUpdated;
-        }
-      }
-    }
+    const envSet = setOfParameterKeysWithWorkload(params, workload);
+    const controllerSet = new Set(envControllerParameters);
+
+    const parametersToRemove = [...envSet].filter(param => !controllerSet.has(param));
+    const parametersToAdd = [...controllerSet].filter(param => !envSet.has(param));
     const exportedValues = getExportedValues(updatedEnvStack);
-    // Return if there's no parameter changes.
-    if (!updated) {
+    // Return if there are no parameter changes.
+    if (parametersToRemove.length + parametersToAdd.length === 0)  {
       return exportedValues;
     }
+
+    for (const param of params) {
```

Comment:
nit: can we rename these to 
```suggestion
    for (const envParam of envParams) {
```